base/applications/rapps/include/rosui.h

```diff
@@ -21,31 +25,34 @@ class CPointerArray
         m_hDpa = DPA_Create(GrowthRate);
     }
 
-    ~CPointerArray()
+    virtual ~CPointerArray()
     {
+        // Here, s_OnRemoveItem() can only call our own OnRemoveItem() implementation...
+        // unless we use the CRTP construct...
         DPA_DestroyCallback(m_hDpa, s_OnRemoveItem, this);
     }
 
 private:
     static INT CALLBACK s_OnRemoveItem(PVOID ptr, PVOID context)
     {
-        CPointerArray * self = (CPointerArray*) context;
-        return (INT) self->OnRemoveItem(reinterpret_cast<T*>(ptr));
+        // TDeriv* pThis = static_cast<TDeriv*>(context);
+        CPointerArray* pThis = (CPointerArray*)context;
+        return (INT)pThis->OnRemoveItem(reinterpret_cast<T*>(ptr));
     }
 
     static INT CALLBACK s_OnCompareItems(PVOID p1, PVOID p2, LPARAM lParam)
     {
-        CPointerArray * self = (CPointerArray*) lParam;
+        CPointerArray* self = (CPointerArray*)lParam;
         return self->OnCompareItems(reinterpret_cast<T*>(p1), reinterpret_cast<T*>(p2));
     }
 
 public:
-    virtual BOOL OnRemoveItem(T * ptr)
+    /*virtual*/ BOOL OnRemoveItem(T* ptr)
```

Comment:
The commenting of `virtual` was part of my attempts at studying the above-mentioned problem.